lnwire/lnwire.go

```diff
@@ -401,6 +401,11 @@ func WriteElement(w io.Writer, element interface{}) error {
 		}
 
 	case DeliveryAddress:
+		// If there is no delivery address, do not write anything.
```

Comment:
I don't think so? That would make messages without the optional delivery address fields fail, wouldn't it?

Reply:
since we don't attempt to write it in case it is zero-length, it should be okay to return an error here.

However see my other comment about zero-length address.

Reply:
Updated to write a zero for empty addresses, as per comment above